src/traces/bar/set_positions.js

```diff
@@ -392,9 +392,11 @@ function setBarCenter(gd, pa, sieve) {
         for(var j = 0; j < calcTrace.length; j++) {
             var calcBar = calcTrace[j];
 
+            // store the actual bar width and position, for use by hover
+            var width = calcBar.w = (barwidthIsArray) ? barwidth[j] : barwidth;
```

Comment:
I suspect we could clean up the hover routine quite a bit more if we stopped using `.x` and `.y` in `calcdata`... then we'd only have to switch at the beginning of the routine "are `xval` and `yval` position and size or vice versa" but I wasn't sure what the consequences of that would be beyond hover, just adding one more value was safe and easy.